Modules/Core/ImageFunction/include/itkRayCastInterpolateImageFunction.hxx

```diff
@@ -1416,10 +1416,17 @@ RayCastInterpolateImageFunction<TInputImage, TCoordRep>::Evaluate(const PointTyp
   ray.ZeroState();
   ray.Initialise();
 
-  PointType origin = this->m_Image->GetOrigin();
-  for (unsigned int i = 0; i < origin.Length; ++i)
-    origin[i] -= 0.5 * this->m_Image->GetSpacing()[i];
-  ray.SetRay(point - origin, direction);
+  const PointType origin = this->m_Image->GetOrigin();
+  const auto      spacing = this->m_Image->GetSpacing();
+
+  PointType rayPosition = point;
+
+  for (unsigned int i = 0; i < PointType::Length; ++i)
+  {
+    rayPosition[i] += 0.5 * spacing[i] - origin[i];
```

Comment:
Subtracting a point `origin` from a vector `spacing` is a little weird mathematically.  How about instead subtracting a vector from a point:
```cpp
rayPosition[i] -= origin[i] - 0.5 * spacing[i];
```

Reply:
@Leengit Thanks for the suggestion. I wouldn't mind following your suggestion, but don't you agree that _in general_ the form

    a += b - c;

Is clearer than the equivalent:

    a -= c - b;

?

Reply:
@Leengit Update: amended the pull request according to your suggestion regarding `rayPosition[i] +/-= ...`